### PR TITLE
Make more callee patterns recognized as "test call callee".

### PR DESCRIPTION
**Pull request by HosokawaR**
## Description

Fixes https://github.com/prettier/prettier/issues/12575

I made these patterns recognized as "test call callee" additionally.

- [test.describe(titile, callback)](https://playwright.dev/docs/api/class-test#test-describe)
- [test.describe.only(title, callback)](https://playwright.dev/docs/api/class-test#test-describe-only)
- [test.describe.parallel(title, callback)](https://playwright.dev/docs/api/class-test#test-describe-parallel)
- [test.describe.parallel.only(title, callback)](https://playwright.dev/docs/api/class-test#test-describe-parallel-only)
- [test.describe.serial(title, callback)](https://playwright.dev/docs/api/class-test#test-describe-serial)
- [test.describe.serial.only(title, callback)](https://playwright.dev/docs/api/class-test#test-describe-serial-only)
- [test.step(title, body)](https://playwright.dev/docs/api/class-test#test-step)

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
What do you think a function like [this](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d53d935951aa815c763fc9441aa452c763294715/rules/utils/is-node-matches.js#L46) ?

File: src/language-js/utils/is-noded-matches.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+/**
+Check if node matches object name or key path.
+
+@param {Node} node - The AST node to check.
+@param {string} nameOrPath - The object name or key path.
+@returns {boolean}
+*/
+function isNodeMatchesNameOrPath(node, nameOrPath) {
+  const names = nameOrPath.trim().split(".");
```
Comment (fisker) [inline src/language-js/utils/is-noded-matches.js:11]: This is used internally, so no need `trim`.

File: src/language-js/utils/is-noded-matches.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  for (let index = names.length - 1; index >= 0; index--) {
+    const name = names[index];
+    if (!name) {
+      return false;
+    }
```
Comment (fisker) [inline src/language-js/utils/is-noded-matches.js:16]: Thi is unnessary too.

File: src/language-js/utils/is-noded-matches.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    if (index === 0) {
+      return (
+        (node.type === "Identifier" && node.name === name) ||
+        (name === "this" && node.type === "ThisExpression")
```
Comment (fisker) [inline src/language-js/utils/is-noded-matches.js:21]: We don't need `this` for now.

File: src/language-js/utils/is-noded-matches.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+function isNodeMatches(node, nameOrPaths) {
+  return nameOrPaths.some((nameOrPath) =>
+    isNodeMatchesNameOrPath(node, nameOrPath)
+  );
+}
+
+module.exports = {
+  isNodeMatchesNameOrPath,
+  isNodeMatches,
+};
```
Comment (fisker) [inline src/language-js/utils/is-noded-matches.js:55]:
```suggestion
module.exports = isNodeMatches;
```

**Review by fisker — CHANGES_REQUESTED** [review]

File: src/language-js/utils/index.js [hunk 5]
```diff
@@ -10,6 +10,7 @@ const {
   getStringWidth,
 } = require("../../common/util.js");
 const { locStart, locEnd, hasSameLocStart } = require("../loc.js");
+const { isNodeMatches } = require("./is-noded-matches.js");
```
Comment (fisker) [inline src/language-js/utils/index.js:13]: File name should be `is-node-matches.js`

**Comment by HosokawaR** [comment]
Thank you for commenting ! 
And, sorry for mistake filename.

I fixed my change following your comment.

File: src/language-js/utils/index.js [hunk 6]
```diff
@@ -475,6 +459,35 @@ function isUnitTestSetUp(node) {
   );
 }
 
+function isTestCallCallee(node) {
+  const patterns = [
```
Comment (fisker) [inline src/language-js/utils/index.js:463]: Should we move this array to outer scope?

Comment (HosokawaR) [reply src/language-js/utils/index.js:463]: I followed [this way](https://github.com/prettier/prettier/blob/b188c905cfaeb238a122b4a95c230da83f2f3226/src/language-js/utils/index.js#L470).

And I thought the smaller the scope of the variable, the better.

However, moving this array to outer scope may be more good performance (?) due to using less memory.

 I want follow your decision.

File: src/language-js/utils/is-node-matches.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
+"use strict";
```
Comment (fisker) [inline src/language-js/utils/is-node-matches.js:1]: Though this function was written by me (fisker), and I'm fine to use it in this repo, but maybe we should add source here? 


```suggestion
"use strict";

// Copied from https://github.com/sindresorhus/eslint-plugin-unicorn/blob/d53d935951aa815c763fc9441aa452c763294715/rules/utils/is-node-matches.js
```

File: src/language-js/utils/index.js (hunk 6 above)
Comment (fisker) [reply src/language-js/utils/index.js:463]: > moving this array to outer scope may be more good performance (?) due to using less memory.

I guess so.  `CallExpression` is so common in ESTree, Let's move it out then.

File: src/language-js/utils/is-node-matches.js (hunk 7 above)
Comment (HosokawaR) [reply src/language-js/utils/is-node-matches.js:1]: Exactly right. I'm sorry.
It lacked consideration.

File: src/language-js/utils/index.js (hunk 6 above)
Comment (HosokawaR) [reply src/language-js/utils/index.js]: I moved .

Please let me know if there are any bad points.

**Review by fisker — APPROVED** [review]
LGTM. Thank you!

**Comment by HosokawaR** [comment]
Thank you for pointing out a lot.
I learned a lot.

File: changelog_unreleased/javascript/12779.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
+#### Make more callee patterns recognized as "test call callee". (#12779 by @HosokawaR)
+
```
Comment (sosukesuzuki) [inline changelog_unreleased/javascript/12779.md:2]: Can you add description for real-world use case? e.g.

```suggestion
Supports test calls like [Playwright `test.describe`](https://playwright.dev/docs/api/class-test#test-describe).
```

File: src/language-js/utils/index.js [hunk 9]
```diff
@@ -475,6 +459,36 @@ function isUnitTestSetUp(node) {
   );
 }
 
+const patterns = [
```
Comment (sosukesuzuki) [inline src/language-js/utils/index.js:462]: `patterns` is too general. What do you think about `testCallCalleePatterns` ?

File: changelog_unreleased/javascript/12779.md (hunk 8 above)
Comment (HosokawaR) [reply changelog_unreleased/javascript/12779.md:2]: Certainly, It's more kind !
I followed your suggestion.

File: src/language-js/utils/index.js (hunk 9 above)
Comment (HosokawaR) [reply src/language-js/utils/index.js]: I agree.
I will use `testCallCalleePatterns`.
Thank you.

**Review by sosukesuzuki — APPROVED** [review]

**Comment by sosukesuzuki** [comment]
Thank you for doing this!

**Comment by dossy** [comment]
Thank you, everyone, for making this PR happen!  😍

**Pull request merged**